pkg/apis/batch/validation/validation.go

```diff
@@ -429,15 +455,23 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 
 // ValidateJobUpdate validates an update to a Job and returns an ErrorList with any errors.
 func ValidateJobUpdate(job, oldJob *batch.Job, opts JobValidationOptions) field.ErrorList {
-	allErrs := apivalidation.ValidateObjectMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta, field.NewPath("metadata"))
+	allErrs := validateJobMetaUpdate(&job.ObjectMeta, &oldJob.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, ValidateJobSpecUpdate(job.Spec, oldJob.Spec, field.NewPath("spec"), opts)...)
 	return allErrs
 }
 
 // ValidateJobUpdateStatus validates an update to the status of a Job and returns an ErrorList with any errors.
```

Comment:
You didn't change this but the comment of this function doesn't seem to match the implementation.

I didn't quite understand why ValidateJobUpdateStatus would impact the metadata?

Reply:
Apparently you can modify metadata when updating objects status. I confirmed this by creating a job, and updating both status & metadata using `kubectl edit job/<job-name> --subresource=status`. For this reason we need to prevent mutation of the label this way.

Note that metadata is also verified for other status updates:
- https://github.com/kubernetes/kubernetes/blob/50bf3a2060ea798600af1a4c125e0b62d08e9680/pkg/apis/apps/validation/validation.go#L237
- https://github.com/kubernetes/kubernetes/blob/50bf3a2060ea798600af1a4c125e0b62d08e9680/pkg/apis/autoscaling/validation/validation.go#L144
- https://github.com/kubernetes/kubernetes/blob/50bf3a2060ea798600af1a4c125e0b62d08e9680/pkg/apis/networking/validation/validation.go#L346

Reply:
Alright.